src/helpers/mergeArrays.js

```diff
@@ -19,7 +19,10 @@ const createMergeArrays = mergeItemFn => (prev, next) => {
 
 const mergeArrays = createMergeArrays((prev, next) => next)
 export const mergeNodeArrays = createMergeArrays((prev, next) => {
-  if (next.chartIDs.every(id => prev.chartIDs.includes(id))) {
+  if (
+    next.chartIDs.length === prev.chartIDs.length &&
+    next.chartIDs.every(id => prev.chartIDs.includes(id))
+  ) {
```

Comment:
why not just `deepEqual(next.chartIDs, prev.chartIDs)`???

Reply:
could be if for example we would get different order of items. But i would agree if you say it's minor and rare case, in this case deepEqual would be good, too

Reply:
btw. i changed it so it returns `prev` in this case.

Ideally maybe we should build do `[...new Set([...prev.chartIDs, ...next.chartIDs])]` and then do deepEq check, in case `next` is a subset of `prev`

Reply:
> could be if for example we would get different order of items. But i would agree if you say it's minor and rare case, in this case deepEqual would be good, too

the `.every => .includes` is `O^2` complexity, the `deepEqual` even if they have a different order, is `On`

Reply:
This function runs only after fetching, shouldn't we treat it as insignificant micro optimisation?
Anyway i switched it to deepEqal